examples_tests/39.DenoiserTonemapper/main.cpp

```diff
@@ -586,152 +584,159 @@ void main()
 				else
 					outParam.denoiserType = EII_NORMAL;
 			}
+			maxResolution[outParam.denoiserType][0] = core::max(maxResolution[outParam.denoiserType][0], outParam.width);
+			maxResolution[outParam.denoiserType][1] = core::max(maxResolution[outParam.denoiserType][1], outParam.height);
 
-			maxResolution[outParam.denoiserType][0] = core::max(maxResolution[outParam.denoiserType][0],outParam.width);
-			maxResolution[outParam.denoiserType][1] = core::max(maxResolution[outParam.denoiserType][1],outParam.height);
 		}
 	}
+	for (size_t i = 0; i < inputFilesAmount; i++)
+	{
+		auto& outParam = images[i];
+		maxResolution[outParam.denoiserType][0] = core::min(maxResolution[outParam.denoiserType][0], tileWidthWithOverlap)*2;
+		maxResolution[outParam.denoiserType][1] = core::min(maxResolution[outParam.denoiserType][1], tileHeightWithOverlap)*2;
+	}
```

Comment:
First of all, just iterate over denoiser types and not input images.

Secondly WHY ARE YOU MULTIPLYING by 2 ?

Reply:
actually it would be good to not clamp `maxResolution` here and use the tile size directly

Reply:
well, size of scratch is currently not calculated properly, and it turns out ~1.3 times smaller than needed